sunpy/coordinates/transformations.py

```diff
@@ -76,108 +63,155 @@ def _observers_are_equal(obs_1, obs_2, string_ok=False):
 # =============================================================================
 
 
-@frame_transform_graph.transform(FunctionTransform, HeliographicStonyhurst,
-                                 HeliographicCarrington)
+def _transform_obstime(frame, obstime):
+    """
+    Transform a frame to a new obstime using the appropriate loopback transformation
+    """
+    if frame.obstime is None:
+        raise ValueError("To perform this transformation the coordinate"
+                         " Frame needs an obstime Attribute")
+
+    # If obstime is None or the obstime matches, nothing needs to be done
+    if obstime is None or np.all(frame.obstime == obstime):
+        return frame
+
+    if isinstance(frame, BaseCoordinateFrame):
+        frame_class = frame.__class__
+    else:  # Assume that it is like SkyCoord and contains the frame
```

Comment:
Is this ever a `SkyCoord` here? I thought the transform graph always operated on frames.

Reply:
Heh, this is a holdover because I was testing this function on both frames and SkyCoords.  I believe you're correct that the latter will never be needed by the machinery.  Removing